In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import os
import shutil
from tqdm import tqdm
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

%matplotlib inline

In [ ]:
%cd /content/drive/MyDrive/KoBART-summarization

/content/drive/MyDrive/KoBART-summarization


In [ ]:
!pip install -r requirements.txt

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |██████████████████████████████▎ | 834.1 MB 1.2 MB/s eta 0:00:42tcmalloc: large alloc 1147494400 bytes == 0x38c28000 @  0x7f89fe855615 0x592b76 0x4df71e 0x59afff 0x515655 0x549576 0x593fce 0x548ae9 0x51566f 0x549576 0x593fce 0x548ae9 0x5127f1 0x598e3b 0x511f68 0x598e3b 0x511f68 0x598e3b 0x511f68 0x4bc98a 0x532e76 0x594b72 0x515600 0x549576 0x593fce 0x548ae9 0x5127f1 0x549576 0x593fce 0x5118f8 0x593dd7
     |████████████████████████████████| 881.9 MB 19 kB/s 
     |████████████████████████████████| 2.5 MB 42.5 MB/s 
     |████████████████████████████████| 813 kB 72.6 MB/s 
     |████████████████████████████████| 8.3 MB 32.3 MB/s 
     |████████████████████████████████| 880 kB 69.7 MB/s 
     |████████████████████████████████| 3.3 MB 45.9 MB/s 
     |████████████████████████████████| 636 kB 72.3 MB/s 
     |████████████████████████████████| 419 kB 70.8 MB/s 
     |█████████████████████

In [ ]:
!pip install torch==1.7.1+cu101 torchvision==0.8.2+cu101 -f https://download.pytorch.org/whl/torch_stable.html

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in links: https://download.pytorch.org/whl/torch_stable.html
     |████████████████████████████████| 735.4 MB 14 kB/s 
     |████████████████████████████████| 12.8 MB 32.0 MB/s 
  Attempting uninstall: torch
    Found existing installation: torch 1.10.0
    Uninstalling torch-1.10.0:
      Successfully uninstalled torch-1.10.0
  Attempting uninstall: torchvision
    Found existing installation: torchvision 0.13.1+cu113
    Uninstalling torchvision-0.13.1+cu113:
      Successfully uninstalled torchvision-0.13.1+cu113
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchtext 0.13.1 requires torch==1.12.1, but you have torch 1.7.1+cu101 which is incompatible.
torchaudio 0.12.1+cu113 requires torch==1.12.1, but you have torch 1.7.1+cu101 which is incompatible.

In [ ]:
!nvidia-smi

Wed Aug 24 14:01:42 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   40C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

In [ ]:
!pip install torchtext==0.8.0

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 6.9 MB 17.6 MB/s 
  Attempting uninstall: torchtext
    Found existing installation: torchtext 0.13.1
    Uninstalling torchtext-0.13.1:
      Successfully uninstalled torchtext-0.13.1


In [ ]:
!pip install torchmetrics==0.6.0

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 329 kB 16.0 MB/s 
  Attempting uninstall: torchmetrics
    Found existing installation: torchmetrics 0.9.3
    Uninstalling torchmetrics-0.9.3:
      Successfully uninstalled torchmetrics-0.9.3


In [ ]:
!pip install pytorch_lightning==1.5.2

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 1.0 MB 26.6 MB/s 
  Attempting uninstall: pyDeprecate
    Found existing installation: pyDeprecate 0.3.0
    Uninstalling pyDeprecate-0.3.0:
      Successfully uninstalled pyDeprecate-0.3.0
  Attempting uninstall: pytorch-lightning
    Found existing installation: pytorch-lightning 1.3.8
    Uninstalling pytorch-lightning-1.3.8:
      Successfully uninstalled pytorch-lightning-1.3.8


In [ ]:
import torch
torch.cuda.empty_cache()

In [ ]:
## train

In [ ]:
!python train.py --gradient_clip_val 1.0 --max_epochs 10 --default_root_dir logs --gpus 1 --batch_size 8 --num_workers 4

Downloading: 100% 4.00/4.00 [00:00<00:00, 2.74kB/s]
Downloading: 100% 111/111 [00:00<00:00, 102kB/s]
Downloading: 100% 682k/682k [00:01<00:00, 597kB/s]
INFO:root:Namespace(accelerator=None, accumulate_grad_batches=None, amp_backend='native', amp_level=None, auto_lr_find=False, auto_scale_batch_size=False, auto_select_gpus=False, batch_size=8, benchmark=False, check_val_every_n_epoch=1, checkpoint_callback=None, checkpoint_path=None, default_root_dir='logs', detect_anomaly=False, deterministic=False, devices=None, enable_checkpointing=True, enable_model_summary=True, enable_progress_bar=True, fast_dev_run=False, flush_logs_every_n_steps=None, gpus=1, gradient_clip_algorithm=None, gradient_clip_val=1.0, ipus=None, limit_predict_batches=1.0, limit_test_batches=1.0, limit_train_batches=1.0, limit_val_batches=1.0, log_every_n_steps=50, log_gpu_memory=None, logger=True, lr=3e-05, max_epochs=10, max_len=512, max_steps=-1, max_time=None, min_epochs=None, min_steps=None, model_path=None, move_m

2 epoch 중간에 runtime 끊어짐

In [ ]:
!pip install pyyaml==5.4.1

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
!python get_model_binary.py --hparams ./logs/tb_logs/default/version_4/hparams.yaml --model_binary ./logs/model_chp/epoch=01-val_loss=1.353.ckpt

get_model_binary.py:13: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  hparams = yaml.load(f)
Downloading: 100% 1.36k/1.36k [00:00<00:00, 1.29MB/s]
Downloading: 100% 496M/496M [00:27<00:00, 17.7MB/s]
Downloading: 100% 4.00/4.00 [00:00<00:00, 3.68kB/s]
Downloading: 100% 111/111 [00:00<00:00, 99.3kB/s]
Downloading: 100% 682k/682k [00:01<00:00, 594kB/s]


In [ ]:
## test

In [ ]:
!pip install git+https://github.com/SKT-AI/KoBART#egg=kobart

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Cloning https://github.com/SKT-AI/KoBART to /tmp/pip-install-0vnovswa/kobart_efcbf4c304e149e6a6b2600f91ad53d4
  Running command git clone -q https://github.com/SKT-AI/KoBART /tmp/pip-install-0vnovswa/kobart_efcbf4c304e149e6a6b2600f91ad53d4
     |████████████████████████████████| 132 kB 33.1 MB/s 
     |████████████████████████████████| 814 kB 54.4 MB/s 
     |████████████████████████████████| 1.9 MB 55.0 MB/s 
     |████████████████████████████████| 596 kB 66.9 MB/s 
     |████████████████████████████████| 79 kB 8.6 MB/s 
     |████████████████████████████████| 9.1 MB 42.9 MB/s 
     |████████████████████████████████| 127 kB 74.1 MB/s 
  Created wheel for kobart: filename=kobart-0.5.1-py3-none-any.whl size=9583 sha256=d0ef661107e44cf5e14e12cf2f088a31624ec61593878c7fec228b62e60bcd25
  Stored in directory: /tmp/pip-ephem-wheel-cache-96_onvx9/wheels/6e/55/c4/bd4fede223bc304089ac8da2a2099a6

In [ ]:
finance_data = pd.read_csv('/content/drive/MyDrive/2022_08_23.csv', encoding='utf-8')

In [ ]:
finance_data['text'][13]

'A street sign for Wall Street is seen in the financial district in New York, U.S., November 8, 2021. REUTERS/Brendan McDermid뉴욕증시가 금리 인상 우려 등으로 큰 폭으로 하락했다. 인플레이션과의 전쟁을 벌이고 있는 연방준비제도(연준·Fed)가 급격한 금리 인상을 조만간 멈출 수 있다는 전망이 지나치게 낙관적인 것이라는 지적이 힘을 얻으면서 투자심리가 위축됐다. 22일(현지시간) 다우지수는 전날보다 643.13포인트(1.91%) 내린 3만3063.61로 마감했다. S&P500지수는 90.49포인트(2.14%) 내린 4137.99로 장을 마쳤다. 나스닥지수는 323.64포인트(2.55%) 내린 1만2381.57로 거래를 마쳤다. 10년물 국채 금리는 3%대를 재돌파했다. 이날 2.989%로 출발한 10년물 수익률은 3.026%로 상승했다. ━"그동안 위험에 너무 안주"...잭슨홀 미팅 앞두고 긴장감↑━이날 다우지수와 S&P500지수는 지난 6월16일 이후 최악의 일일 하락폭을 기록했고, 나스닥지수도 6월28일 이후 가장 큰 폭으로 떨어졌다. 최근 월스트리트 일각에선 인플레이션이 정점을 찍으면서 매파적 기조의 연준이 금리 인상 속도를 늦추고, 내년에는 비둘기파적 기조로 방향을 전환할 수 있을 것이라는 기대감이 고조됐다. 하지만 최근 연준 관계자들이 잇따라 강경한 발언을 이어가면서 시장에는 다시 긴장하는 모습이다. 이번주 목요일부터 열리는 잭슨홀 미팅을 앞두고 시장은 촉각을 곤두세우고 있다. 글렌메드의 마이클 레이놀즈 투자전략 부사장은 마켓워치에 "시장은 거시경제 환경에 대한 위험에 너무 안주해왔다"며 "우리는 향후 12개월 동안 경기 침체 위험이 50%에 달할 것으로 보고 있다"고 진단했다. 이어 "우리가 처한 상황을 볼 때 시장은 이런 밸류에이션 속에서 약간 과열된 것처럼 보이며, 우리는 주식에 대한 비중축소 의견을 계속 유지한다"고 밝혔다. 레이놀즈 부사장은 "금

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

summarization = []

for i in range(len(finance_data['text'])):
  text = finance_data['text'][i][:2200]
  input_ids = tokenizer.encode(text)
  input_ids = torch.tensor(input_ids)
  input_ids = input_ids.unsqueeze(0)
  output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
  output = tokenizer.decode(output[0], skip_special_tokens=True)
  summarization.append(output)

In [ ]:
dataframe = pd.DataFrame({"url": finance_data['url'], "title": finance_data['title'], "text": finance_data['text'],"생성요약": summarization})

In [ ]:
dataframe

url  \
0   https://finance.naver.com/news/news_read.naver...   
1   https://finance.naver.com/news/news_read.naver...   
2   https://finance.naver.com/news/news_read.naver...   
3   https://finance.naver.com/news/news_read.naver...   
4   https://finance.naver.com/news/news_read.naver...   
5   https://finance.naver.com/news/news_read.naver...   
6   https://finance.naver.com/news/news_read.naver...   
7   https://finance.naver.com/news/news_read.naver...   
8   https://finance.naver.com/news/news_read.naver...   
9   https://finance.naver.com/news/news_read.naver...   
10  https://finance.naver.com/news/news_read.naver...   
11  https://finance.naver.com/news/news_read.naver...   
12  https://finance.naver.com/news/news_read.naver...   
13  https://finance.naver.com/news/news_read.naver...   

                                             title  \
0     잭슨 홀 리스크·유럽 침체우려에도 나스닥 소폭 상승 출발 [데일리 국제금융시장]   
1                [Asia마감] 긴축 공포에 짓눌린 시장…닛케이 1.19%↓   
2    [마켓뷰] 코스피·코스닥 1% 넘게 하락…치솟는 美 달러화 가치에 투자금 ‘썰물’   
3               [마감시황]코스피, 강달러 지속에 1%대 하락…2430선 마감   
4                          [유럽개장]장 초반 하락세…獨 0.35%↓   
5   [시황종합] 코스피, 강달러 충격에 5일째 하락 2435선 마감…환율 1345.5원   
6        [외환마감]환율, 1340원대 연고점 경신…尹대통령 언급+당국개입도 역부족   
7              [코스닥 마감]킹달러·잭슨홀 경계감에 1.5%↓…780선 턱걸이   
8             [Asia오전] 美 추락에 亞증시도 '뚝뚝뚝'…닛케이 1.17%↓   
9            [외환브리핑]109선 찍은 달러인덱스…환율, 1340원대 안착 시도   
10     [굿모닝 증시]잭슨홀 경계·강달러에 약세 보인 美 증시…코스피 하락 출발 예상   
11                      [일일펀드동향] 채권형펀드에서 911억원 순유출   
12    금리인상 우려에 美증시 '큰폭 하락'…환율 급등 "금융위기 수준" [모닝브리핑]   
13           "위험 과소평가" 잭슨홀 앞두고 긴장↑...美증시 '뚝'[뉴욕마감]   

                                                 text  \
0   다우 0.065%·S&P 0.069% 상승10년 물 국채 3.06%로 올라유로존 8...   
1   23일 아시아 주요국 증시가 일제히 하락 마감했다. 통화긴축, 경기침체에 대한 우려...   
2   유럽, 500년 만에 최악의 가뭄…경기 우려 심화유로화 가치 20년 만에 최저치로,...   
3   [서울=뉴시스] 조성우 기자 =코스피가 전 거래일(2462.50)보다 27.16포인...   
4   23일 유럽 증시가 장 초반 하락세를 보이고 있다.이날 오후 4시20분(한국시간) ...   
5   기관, 6거래일 간 1조6541억원 '팔자…외인, 선물 1926억원 순매도대형주 부...   
6   당국개입에 1330원대 하락 이후 오후 다시 상승 흐름1346원선까지 오르면서 13...   
7   원·달러 환율 장중 1342원에 투자심리 위축기관 나홀로 '팔자'…외인·개인 '사자...   
8   22일 오전 아시아 증시는 모두 하락세를 나타냈다. 통화긴축, 경기침체 대한 우려가...   
9   글로벌 달러인덱스 109선 급등, 달러 초강세제롬 파월 의장 잭슨홀미팅 3일 앞두고...   
10  [이미지출처=로이터연합뉴스]23일 국내 증시는 잭슨홀 미팅을 앞두고 하락 마감한 미...   
11  미국의 긴축 기조가 유지될 것이라는 전망이 나오는 가운데, 채권형 펀드 시장에서는 ...   
12  ◆ '잭슨홀 회의' 앞두고 공격적 긴축 예상 확산미 중앙은행(연방준비제도·Fed)의...   
13  A street sign for Wall Street is seen in the f...   

                                                 생성요약  
0   제롬 파월 미 연방준비제도 의장이 잭슨 홀 미팅 연설에서 매파적인 모습을 보일 수 ...  
1   일본 도쿄 주식시장에서 닛케이225지수는 전날 대비 341.75포인트(1.19%) ...  
2   코스피지수가 전 거래일보다 27.16포인트(1.1%) 내린 2435.34로 마감해 ...  
3   코스피가 전 거래일보다 27.16포인트(1.10%) 내린 2435.34에 거래를 마...  
4          23일 유럽 증시가 장 초반 하락세를 보이며 장 초반 하락세를 보이고 있다.  
5   23일 코스피지수가 달러 강세의 영향으로 1%가량 하락하며 2435선에서 마감했고 ...  
6   글로벌 달러 강세에 기인한 원·달러 환율 상승 과정에서 역외 등을 중심으로 한 투기...  
7   코스닥 지수가 주요국 중앙은행 총재들이 모이는 잭슨홀 미팅을 앞두고 경계감이 작동하...  
8   22일 오전 도쿄증시의 닛케이225지수는 전일 대비 337.58포인트(1.17%) ...  
9   글로벌 위험회피 심리와 안전자산인 달러화 강세 흐름이 이어지고 있는 뉴욕증시는 연준...  
10  23일 국내 증시는 잭슨홀 미팅을 앞두고 하락 마감한 미국 증시의 영향으로 하락 출...  
11  22일 금융투자협회에 따르면 미국의 긴축 기조가 유지될 것이라는 전망이 나오는 가운...  
12  중앙은행의(연방준비제도·Fed)의 매파적 통화정책이 계속될 것이란 우려가 커지면서 ...  
13  인플레이션과의 전쟁을 벌이고 있는 연방준비제도(연준·Fed)가 급격한 금리 인상을 ...

In [ ]:
dataframe.to_csv('/content/drive/MyDrive/2022_08_23_summarization.csv')

In [ ]:
## external test dataset

In [ ]:
finance_data['text'][0]

'다우 0.065%·S&P 0.069% 상승10년 물 국채 3.06%로 올라유로존 8월 PMI 49.2 2달 연속 50 아래[서울경제] 제롬 파월 미 연방준비제도(Fed·연준) 의장이 잭슨 홀 미팅 연설에서 매파적인 모습을 보일 수 있다는 우려가 커지는 가운데, 유럽의 경기침체 가능성이 크게 높아지면서 미국 뉴욕증시의 주요 지수가 출발했다.23일(현지 시간) 미 경제 방송 CNBC에 따르면 이날 오전9시35분 현재 뉴욕증권거래소(NYSE)의 다우존스산업평균지수가 전날보다 21.58포인트(0.065%) 오른 3만3085.19에 거래되고 있다.스탠더드앤드푸어스(S&P)500이 2.87포인트(0.069%) 상승한 4140.86, 나스닥은 23.77포인트(0.18%) 오른 1만2405.34를 기록 중이다.CNBC는 “금리 인상 우려 속에 여름랠리가 약해지고 10년 만기 미 국채금리가 3%를 웃돌고 있다”고 전했다. 이날 오전 10년 만기 미 국채금리가 연 3.06%선까지 올랐다. 다만, 전날 하락한 데 따른 매수 수요가 장초반에 몰린 것으로 보인다.리사 샬렛 모건스탠리 웰스 매니지먼트의 글로벌 투자위원회 헤드는 “월가가 인플레이션과 커지는 경기침체 리스크 그리고 언젠가 내려와야만 하는 어닝 기대치를 과소평가하고 있다”며 “베어마켓은 끝나지 않았다”고 지적했다.유럽의 침체 가능성도 커지고 있다. S&P글로벌은 유로존 8월 종합 구매관리자지수(PMI)가 49.2로 7월(49.9)에 이어 2개월 연속 50을 밑돌았다고 밝혔다. 50 밑은 위축, 초과는 확장을 뜻한다.#미국 경제와 월가, 연준에 대한 깊이 있는 분석을 제공하는 ‘김영필의 3분 월스트리트’가 유튜브 채널을 통해 여러분을 찾아갑니다. 매주 화~토 오전6시55분 서울경제 ‘어썸머니’ 채널에서 생방송합니다. ‘3분 월스트리트’ 기사에 관한 보다 상세한 설명을 들을 수 있으니 많은 관심 부탁드립니다.#기자페이지를 구독하시면 미국 경제와 월가의 뉴스를 쉽게 받아보실 수 있습니다.\n'

In [ ]:
finance_data['title'][0]

'잭슨 홀 리스크·유럽 침체우려에도 나스닥 소폭 상승 출발 [데일리 국제금융시장]'

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

다우 0.065%·S&P 0.069% 상승10년 물 국채 3.06%로 올라유로존 8월 PMI 49.2 2달 연속 50 아래[서울경제] 제롬 파월 미 연방준비제도(Fed·연준) 의장이 잭슨 홀 미팅 연설에서 매파적인 모습을 보일 수 있다는 우려가 커지는 가운데, 유럽의 경기침체 가능성이 크게 높아지면서 미국 뉴욕증시의 주요 지수가 출발했다.23일(현지 시간) 미 경제 방송 CNBC에 따르면 이날 오전9시35분 현재 뉴욕증권거래소(NYSE)의 다우존스산업평균지수가 전날보다 21.58포인트(0.065%) 오른 3만3085.19에 거래되고 있다.스탠더드앤드푸어스(S&P)500이 2.87포인트(0.069%) 상승한 4140.86, 나스닥은 23.77포인트(0.18%) 오른 1만2405.34를 기록 중이다.CNBC는 “금리 인상 우려 속에 여름랠리가 약해지고 10년 만기 미 국채금리가 3%를 웃돌고 있다”고 전했다. 이날 오전 10년 만기 미 국채금리가 연 3.06%선까지 올랐다. 다만, 전날 하락한 데 따른 매수 수요가 장초반에 몰린 것으로 보인다.리사 샬렛 모건스탠리 웰스 매니지먼트의 글로벌 투자위원회 헤드는 “월가가 인플레이션과 커지는 경기침체 리스크 그리고 언젠가 내려와야만 하는 어닝 기대치를 과소평가하고 있다”며 “베어마켓은 끝나지 않았다”고 지적했다.유럽의 침체 가능성도 커지고 있다. S&P글로벌은 유로존 8월 종합 구매관리자지수(PMI)가 49.2로 7월(49.9)에 이어 2개월 연속 50을 밑돌았다고 밝혔다. 50 밑은 위축, 초과는 확장을 뜻한다.#미국 경제와 월가, 연준에 대한 깊이 있는 분석을 제공하는 ‘김영필의 3분 월스트리트’가 유튜브 채널을 통해 여러분을 찾아갑니다. 매주 화~토 오전6시55분 서울경제 ‘어썸머니’ 채널에서 생방송합니다. ‘3분 월스트리트’ 기사에 관한 보다 상세한 설명을 들을 수 있으니 많은 관심 부탁드립니다.#기자페이지를 구독하시면 미국 경제와 월가의 뉴스를 쉽게 받아보실 수 있습니다.\n
제롬 파월 미 연방준비제도 의장이 잭

모델 출력: 제롬 파월 미 연방준비제도 의장이 잭슨 홀 미팅 연설에서 매파적인 모습을 보일 수 있다는 우려가 커지는 가운데, 유럽의 경기침체 가능성이 크게 높아지면서 미국 뉴욕증시의 주요 지수가 출발했다.

제목: 잭슨 홀 리스크·유럽 침체우려에도 나스닥 소폭 상승 출발 [데일리 국제금융시장]

In [ ]:
finance_data['text'][1]

'23일 아시아 주요국 증시가 일제히 하락 마감했다. 통화긴축, 경기침체에 대한 우려가 되살아나면서 성장주 중심의 매도세가 이어진 여파다.이날 일본 도쿄 주식시장에서 닛케이225지수는 전날 대비 341.75포인트(1.19%) 하락한 2만8452.75로 장을 마감했다. 나흘 연속 하락한 것이다. 토픽스지수는 21.15포인트(1.06%) 내려간 1971.44를 기록했다.니혼게이자이신문에 따르면 전날 미국 주식시장에서 주요 지수가 하락함에 따라 도쿄 주식시장에서도 운용 리스크를 회피하고 싶은 투자자들의 매도가 우세했다.닛케이지수 평균 하락폭은 장중 한 때 400포인트에 다다랐다.중국 본토 상하이종합지수는 전장대비 1.57포인트(0.05%) 내려간 3276.22에, 대만 자취엔지수는 149.25포인트(0.98%) 내려간 1만5095.89에 마감했다. 홍콩 항셍지수는 현지시간 이날 오후 3시26분 기준 전장대비 226.56포인트(1.15%) 빠진 1만9430.42를 가리키고 있다.  시장은 오는 26일(현지시간) 예정된 제롬 파월 미 연방준비제도(Fed, 연준) 의장의 잭슨홀 미팅 연설에 촉각을 곤두세우고 있다. 최근들어 연준 고위인사들은 부쩍 긴축 관련 언급 수위를 높이고 있고, 달러화 가치는 다시 폭등하는 추세다. \n'

In [ ]:
finance_data['title'][1]

'[Asia마감] 긴축 공포에 짓눌린 시장…닛케이 1.19%↓'

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

23일 아시아 주요국 증시가 일제히 하락 마감했다. 통화긴축, 경기침체에 대한 우려가 되살아나면서 성장주 중심의 매도세가 이어진 여파다.이날 일본 도쿄 주식시장에서 닛케이225지수는 전날 대비 341.75포인트(1.19%) 하락한 2만8452.75로 장을 마감했다. 나흘 연속 하락한 것이다. 토픽스지수는 21.15포인트(1.06%) 내려간 1971.44를 기록했다.니혼게이자이신문에 따르면 전날 미국 주식시장에서 주요 지수가 하락함에 따라 도쿄 주식시장에서도 운용 리스크를 회피하고 싶은 투자자들의 매도가 우세했다.닛케이지수 평균 하락폭은 장중 한 때 400포인트에 다다랐다.중국 본토 상하이종합지수는 전장대비 1.57포인트(0.05%) 내려간 3276.22에, 대만 자취엔지수는 149.25포인트(0.98%) 내려간 1만5095.89에 마감했다. 홍콩 항셍지수는 현지시간 이날 오후 3시26분 기준 전장대비 226.56포인트(1.15%) 빠진 1만9430.42를 가리키고 있다.  시장은 오는 26일(현지시간) 예정된 제롬 파월 미 연방준비제도(Fed, 연준) 의장의 잭슨홀 미팅 연설에 촉각을 곤두세우고 있다. 최근들어 연준 고위인사들은 부쩍 긴축 관련 언급 수위를 높이고 있고, 달러화 가치는 다시 폭등하는 추세다. \n
일본 도쿄 주식시장에서 닛케이225지수는 전날 대비 341.75포인트(1.19%) 하락한 2만8452.75로 장을 마감했으며 중국 본토 상하이종합지수는 전장대비 1.57포인트(0.05%) 내려간 3276.22에, 대만 자취엔지수는 149.25포인트(0.98%) 내려간 1만5095.89에 마감했다.


모델 출력: 일본 도쿄 주식시장에서 닛케이225지수는 전날 대비 341.75포인트(1.19%) 하락한 2만8452.75로 장을 마감했으며 중국 본토 상하이종합지수는 전장대비 1.57포인트(0.05%) 내려간 3276.22에, 대만 자취엔지수는 149.25포인트(0.98%) 내려간 1만5095.89에 마감했다.

제목: [Asia마감] 긴축 공포에 짓눌린 시장…닛케이 1.19%↓

In [ ]:
finance_data['text'][2]

'유럽, 500년 만에 최악의 가뭄…경기 우려 심화유로화 가치 20년 만에 최저치로, 강달러 심화        23일 코스피지수가 1% 넘게 하락했다. 미 달러화 가치가 초강세를 지속하고 있고, 유럽의 경기 침체에 대한 우려가 커지자, 우리 증시도 타격을 입은 것으로 해석된다.23일 오후 서울 중구 하나은행 딜링룸 전광판에 코스피지수, 미 달러화에 대한 원화 환율이 표시돼 있다. /연합뉴스        이날 코스피지수는 전 거래일보다 27.16포인트(1.1%) 내린 2435.34로 마감했다. 장 초반까지만 해도 낙폭이 크지 않았으나 오전 11시를 기점으로 2430대로 떨어진 뒤 박스권 안에서 지지부진한 흐름을 지속했다.유가증권시장 현물시장에서 국내 기관은 1313억원을 순매도했다. 선물시장에서는 외국인의 매도세가 강했다. 외국인은 이날 코스피200 선물을 1925억원 순매도한 것으로 집계됐다.국내 기관이 하루 동안 가장 많이 판 종목은 삼성전자였다. 828억원을 순매도했다. 기관 투자자들은 이달 들어서만 삼성전자 주식을 총 7715억원어치 팔았다. 그중 연기금의 순매도액이 절반을 넘는 4367억원에 달했다.반면 개인은 이날 유가증권시장 현물 시장에서 1383억원을, 선물 시장에서 1408억원을 순매수했다. 개인 투자자들은 지난 17일부터 5거래일 연속 ‘사자’ 행렬을 이어가고 있다.코스피뿐 아니라 코스닥지수도 큰폭으로 떨어졌다. 이날 코스닥지수는 전 거래일보다 12.45포인트(1.56%) 내린 783.42로 마감했다.코스닥지수의 하락 역시 국내 기관이 이끌었다. 국내 기관은 이날 코스닥시장에서 총 632억원어치를 팔았다. 지난 12일부터 7거래일째 순매도를 지속하고 있는데, 이 기간 누적 매도액이 6400억원을 넘는다.이날 우리 증시의 하락은 지난 밤 미 뉴욕 증시의 급락에 영향을 받은 것으로 풀이된다. 뉴욕 3대 주가지수는 유럽의 가뭄과 경기 침체 우려, 미 달러화 강세 영향으로 급락했다.22일(현지 시각) 뉴욕증권거래소(NYSE)에서 다우존스30산업평균지수

In [ ]:
finance_data['title'][2]

'[마켓뷰] 코스피·코스닥 1% 넘게 하락…치솟는 美 달러화 가치에 투자금 ‘썰물’'

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

유럽, 500년 만에 최악의 가뭄…경기 우려 심화유로화 가치 20년 만에 최저치로, 강달러 심화        23일 코스피지수가 1% 넘게 하락했다. 미 달러화 가치가 초강세를 지속하고 있고, 유럽의 경기 침체에 대한 우려가 커지자, 우리 증시도 타격을 입은 것으로 해석된다.23일 오후 서울 중구 하나은행 딜링룸 전광판에 코스피지수, 미 달러화에 대한 원화 환율이 표시돼 있다. /연합뉴스        이날 코스피지수는 전 거래일보다 27.16포인트(1.1%) 내린 2435.34로 마감했다. 장 초반까지만 해도 낙폭이 크지 않았으나 오전 11시를 기점으로 2430대로 떨어진 뒤 박스권 안에서 지지부진한 흐름을 지속했다.유가증권시장 현물시장에서 국내 기관은 1313억원을 순매도했다. 선물시장에서는 외국인의 매도세가 강했다. 외국인은 이날 코스피200 선물을 1925억원 순매도한 것으로 집계됐다.국내 기관이 하루 동안 가장 많이 판 종목은 삼성전자였다. 828억원을 순매도했다. 기관 투자자들은 이달 들어서만 삼성전자 주식을 총 7715억원어치 팔았다. 그중 연기금의 순매도액이 절반을 넘는 4367억원에 달했다.반면 개인은 이날 유가증권시장 현물 시장에서 1383억원을, 선물 시장에서 1408억원을 순매수했다. 개인 투자자들은 지난 17일부터 5거래일 연속 ‘사자’ 행렬을 이어가고 있다.코스피뿐 아니라 코스닥지수도 큰폭으로 떨어졌다. 이날 코스닥지수는 전 거래일보다 12.45포인트(1.56%) 내린 783.42로 마감했다.코스닥지수의 하락 역시 국내 기관이 이끌었다. 국내 기관은 이날 코스닥시장에서 총 632억원어치를 팔았다. 지난 12일부터 7거래일째 순매도를 지속하고 있는데, 이 기간 누적 매도액이 6400억원을 넘는다.이날 우리 증시의 하락은 지난 밤 미 뉴욕 증시의 급락에 영향을 받은 것으로 풀이된다. 뉴욕 3대 주가지수는 유럽의 가뭄과 경기 침체 우려, 미 달러화 강세 영향으로 급락했다.22일(현지 시각) 뉴욕증권거래소(NYSE)에서 다우존스30산업평균지수는

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

윤석열 대통령에 이어 외환·금융당국도 구두개입에 나섰지만 1350원 턱 밑까지 차오른 원·달러 환율이 좀처럼 진정될 기미를 보이지 않고 있다. 외환당국 관계자는 23일 오전 9시22분께 "최근 글로벌 달러 강세에 기인한 원달러 환율 상승 과정에서 역외 등을 중심으로 한 투기적 요인이 있는지에 대해 면밀히 점검해 나갈 것"이라고 말했다. 외환당국이 공식 구두개입에 나선 건 지난 3월7일, 4월25일, 6월13일 이후 네번째다. 이날 구두개입 직후 서울 외환시장에서 환율은 1337.0원까지 하락하면서 1340원대 아래로 내려갔다. 하지만 다시 하락폭을 일부 반납한 뒤 전거래일(1339.8원)보다 5.7원 오른 1345.5원에 거래를 마쳤다. 윤석열 대통령도 이날 오전 서울 용산 대통령청사 출근길 약식 회견에서 "국민 여러분이 1340원까지 치솟은 환율 때문에 걱정이 많을 것 같다"며 "달러화 강세와 원화 약세의 통화 상황이 우리 시장에 부정적인 영향을 미치지 않도록 비상경제 민생회의 등을 통해 리스크 관리를 잘 해가겠다"고 밝힌 바 있다. 이복현 금융감독원장의 경우 이날 열린 임원회의에서 "자본시장 변동성이 확대되는 경우 이를 악용해 투자자 등 피해를 야기하고 시장 신뢰성을 저해하며 시장 변동성을 더욱 확대시키는 불법·불공정행위가 발생할 수 있다"고 언급했다. 이 원장은 또 "이러한 자본시장 불법·불공정행위에 대해 금융위 자본시장조사단과 공조해 철저히 조사하고, 발견된 위법행위에 대해서는 엄중 조치할 계획"이라고 강조했다. 이와 함께 "금융회사는 금융불안 지속에 대비해 영업에 내실을 기할 필요가 있다"며 "투자자도 시장 내 과민반응에 따른 쏠림 현상과 과도한 레버리지를 경계할 필요가 있다"고 덧붙였다. 다만 정부가 환율 안정을 위해 과도한 외환 시장 개입에 나설 가능성은 낮다. 추경호 경제부총리 겸 기획재정부 장관은 지난 19일 한 방송에서 "시장 흐름을 지켜보고 있다"면서도 "너무 걱정 안 해도 될 것 같다"고 했다. 방기선 기획재정부 1차관도 같은날 비상경제차

모델 출력: 윤석열 대통령에 이어 외환·금융당국도 구두개입에 나섰지만 1350원 턱 밑까지 차오른 원·달러 환율이 좀처럼 진정될 기미를 보이지 않고 있으며 외환당국 관계자는 23일 오전 9시22분께 "최근 글로벌 달러 강세에 기인한 원달러 환율 상승 과정에서 투기적 요인이 있는지에 대해 면밀히 점검해 나갈 것"이라고 말했다.

정답: 尹에 이어 외환·금융당국 경고에도…고삐 풀린 환율

In [ ]:
## internal test dataset

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

기다리던 새해가 밝았네요.\n  올해 제가 드디어 아빠가 됩니다.\n \" 국내에서 처음으로 '남의 팔'을 이식받은 손진욱(39) 씨의 2020년은 특별하다.\n  지난해부터 기다리던 그의 2세가 오는 8월 태어나기 때문이다.\n  장애인으로 살던 그가 팔 이식 수술 후 결혼에 골인했고, 이어 행복한 가정을 꾸려 새해 아빠가 된다.\n   손씨는 2018년 가을 지인의 소개로 만나 교제해 온 여자친구와 지난해 6월 대구에서 부부의 연을 맺었다.\n  이후 운수업, 서비스업 등 평범한 30대 직장인으로 생활해왔다.\n  그는 \"장애가 있던 시절에는 직장 다니고, 아이 키우는 평범한 부부를 볼 때 제일 부러웠다\"며\"올해는 아빠가 되는 꿈을 이루는 해\"라고 말했다.\n   손씨는“팔 이식 수술을 받기 위해 병원에 입원한 2017년 1월(수술은 2월)이 첫 번째 새 인생을 시작한 해라면 지난해 결혼이 두 번째 인생, 2020년은 2세가 태어나는 올해는 세 번째 새로운 인생\"이라고 했다.\n    그는 지난 2015년 공장에서 일하다 사고로 왼쪽 팔을 잃었다.\n  의수(義手)를 끼고 생활하다 2017년 2월 대구 W병원과 영남대병원에서 뼈와 신경·근육·혈관 등이 포함된 다른 사람 팔을 이식받았다.\n   새해가 그에게 특별한 이유는 하나 더 있다.\n  수술한 팔이 점점 더 정상인의 팔에 가까워지고 있어서다.\n  희망과 행복이라는 말의 뜻을 이젠 이해하겠다는 손씨는 “수술 3년 차를 맞으면서 ‘남의 팔’이 이젠 진짜 내 팔이 된 것 같다”고 했다.\n  그는 수술 초기에는 팔과 손이 저리고 시려지는 고통을 겪었다.\n  컴퓨터나 스마트폰 자판을 두드리는 것 같은 정교한 움직임은 어려웠다.\n  “왼쪽 팔이 달렸지만, 내 것이 아니라 남의 것이구나”하는 생각이 들 정도였다고 한다.\n   수술 몇 달이 지나면서 그의 몸은 조금씩 남의 팔을 받아들였다.\n  5개 손가락 모두 움직일 수 있게 되고 손에 힘을 줄 수 있었다.\n  야구공을 쥐고 던지거나 다른 사

모델 출력: 장애인으로 살던 손 씨는 팔 이식 수술 후 결혼에 골인했고 행복한 가정을 꾸려 새해 아빠가 된다.

정답: 공장에서 일하다 왼쪽 팔을 잃은 그는 팔 이식 수술 후 결혼에 성공하여 아빠가 되는 꿈을 이루었다.

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

김정은 국무위원장이 지난달 28일부터 31일까지 열린 노동당 중앙위원회 전원회의 보고에서 “곧 머지않아 조선민주주의인민공화국이 보유하게 될 새로운 전략무기를 목격하게 될 것”이라고 말했다고 조선중앙통신이 1일 전했다.\n  통일부는 이날 논평에서 “이(새 전략무기 공개)를 행동으로 옮길 경우 비핵화 협상과 한반도 평화정착 노력에 도움이 되지 않는다”고 우려했다.\n  그렇다면 김 위원장이 언급한 새로운 전략무기는 과연 무엇일까.\n     전략무기는 핵무기나 핵탄두를 장거리로 투사하는 무기를 뜻한다.\n  전문가들은 북한의 새 전략무기를 크게 3가지로 예상한다.\n   ①신형 전략잠수함= 정부 소식통은 “군 당국이 제일 먼저 떠올린 북한의 새 전략무기는 신형 전략잠수함”이라고 전했다.\n  전략잠수함은 핵탄두를 단 잠수함발사탄도미사일(SLBM)을 싣고 다닌다.\n  북한은 SLBM을 ‘수중전략탄도탄’이라고 부른다.\n  권용수 전 국방대 교수는 “전략잠수함과 SLBM은 탐지가 어렵기 때문에 미국이 경계한다”며 “북한이 전략잠수함에서 SLBM을 발사하면 미국은 큰 충격을 받을 것”이라고 예상했다.\n    북한은 이미 전략잠수함과 SLBM을 보유하고 있다.\n  2016년 8월 24일 북한의 고래급 잠수함(2000t급)이 SLBM인 북극성-1형을 성공적으로 발사했다.\n  이어 지난해 10월 2일엔 북극성-1형보다 더 멀리 날아가는 북극성-3형을 수중 바지선에서 쐈다.\n    문근식 한국국방안보포럼 대외협력국장은 “1발의 SLBM만 갖고 다니는 고래급은 SLBM 시험발사 전용”이라면서 ”북한은 고래급보다 더 커 여러 발의 SLBM을 탑재하는 전략잠수함이 필요하다“고 설명했다.\n   북한의 관영 매체는 지난해 7월 23일 신포 조선소에서 김 위원장이 신형 잠수함을 둘러보는 사진을 공개했다.\n  사진 속 잠수함은 3000t은 넘어 보인다.\n  군 관계자는 “크기와 모양이 함교에 3발의 SLBM을 넣는 옛 소련의 골프급과 닮았다”고 말했다.\n    한 정부 

모델 출력: 김 국무 국무위원장이 노동당 중앙위원회 전원회의 보고에서 조선민주주의인민공화국이 보유하게 될 새로운 전략무기를 목격하게 될 것이라고 말했다고 조선중앙통신이 1일 전했다.

정답: 노동당 중앙위원회 전원회의 보고에서 북한의 김 국무위원장이 새로운 전략무기를 목격하게 될 것이라고 말해 전문가들은 전략무기를 3가지로 추측했다.

In [ ]:
import torch
from transformers import PreTrainedTokenizerFast
from transformers.models.bart import BartForConditionalGeneration

model = BartForConditionalGeneration.from_pretrained('./kobart_summary')
tokenizer = PreTrainedTokenizerFast.from_pretrained('gogamza/kobart-base-v1')

text = input()

if text:
    input_ids = tokenizer.encode(text)
    input_ids = torch.tensor(input_ids)
    input_ids = input_ids.unsqueeze(0)
    output = model.generate(input_ids, eos_token_id=1, max_length=512, num_beams=5)
    output = tokenizer.decode(output[0], skip_special_tokens=True)
    print(output)

신 의원은 \"총선에 제가 다시 도전한다는 의사표시로 예비후보를 등록했다\"고 말했다.\n  다른 예비후보자 2명은 경기 의왕시 내손로 건물에 선거사무소를 차리고 현수막을 내걸었다.\n  신 의원 사무소도 같은 건물에 있다.\n  신 의원은 \"저만 현수막이 없으면 이상할 것 아닌가.\n  사람들이 (제가 나오지 않는다고) 오해할 수도 있어서 현수막을 걸었다\"고 말했다.\n   천정배 대안신당 의원도 지역구 광주 서을에서 예비후보 등록을 마쳤다.\n  천 의원은 양향자 민주당 전 최고위원 등 여당 소속 출마자들의 거센 도전을 받고 있다.\n  천 의원은 \"연초에 배우자가 인사를 다닐 때 (예비후보 등록 전에 선거운동을 했다고) 선거법상 시비를 (상대후보가) 걸 가능성이 있어서 미리 등록한 것\"이라고 말했다.\n  한국당 비례대표인 윤종필 의원은 김병관 민주당 의원의 지역구인 경기 성남 분당갑에 출사표를 던졌다.\n  윤 의원은 2017년부터 이 지역에서 활동해왔고 현재 한국당 성남 분당갑 당협위원장을 맡고 있다.\n  선거운동에 일찍 돌입해 본선에서 상대 당 현역 의원을 꺾겠다는 의지로 보인다.\n  '서울 강남·PK 험지 정면돌파'부산에 지역구를 둔 민주당 현역 의원 3명도 나란히 예비후보 명단에 이름을 올렸다.\n  초선인 전재수(부산 북-강서갑), 박재호(부산 남을), 최인호(부산 사하갑) 의원 등이다.\n  이들 셋은 ‘조국 사태’ 이후 PK(부산·경남) 민심이 여당에 호의적이진 않다고 보고 있다.\n  서둘러 움직이지 않으면 안 된다는 분위기가 읽혔다고 한다.\n  박재호 의원은 \"우리처럼 '적지'에 있는 의원들은 무조건 예비후보를 등록하는 게 순서인 것 같다\"며 \"지난해 6월 지방선거에 비하면 부산이 어렵다 보니 서둘러 선거모드에 돌입한 측면도 있다\"고 했다.\n  전재수 의원은 \"현역 의원이 선거기간에 누릴 수 있는 몇 가지 기득권이 있는데 나는 거기에 기대지 않겠다는 의지가 있었다\"고 말했다.\n  민주당 비례대표인 박경미 의원은

모델 출력: 신 의원은 경기 의왕시 내손로 건물에 선거사무소를 차리고 현수막을 걸었으며 다른 예비후보자 2명은 경기 의왕시 내손로 건물에 선거사무소를 차리고 현수막을 걸었다.

정답: 총선에 도전하는 예비후보자들이 선거사무소를 차리고 현수막을 걸자 신 의원도 사람들에게 예비후보를 등록한 것을 보여주기 위해 나섰다.

In [ ]:
## ref. https://younghwani.github.io/posts/kobart-summary-3/